pkg/asset/installconfig/aws/availabilityzones.go

```diff
@@ -92,24 +92,28 @@ func filterZonesByType(ctx context.Context, session *session.Session, region str
 
 // availabilityZones retrieves a list of zones type 'availability-zone' in the region.
 func availabilityZones(ctx context.Context, session *session.Session, region string) ([]string, error) {
-	return filterZonesByType(ctx, session, region, typesaws.AvailabilityZoneType)
+	return filterZonesByType(ctx, session, region, typesaws.AvailabilityZoneType, true)
```

Comment:
rather than adding a new var to the function, could you just check for a nil slice in the error scenario?

```suggestion
	localZones, err := filterZonesByType(ctx, session, region, typesaws.LocalZoneType)
	if localZones == nil {
	  return some-error
	}
```